### PR TITLE
`no-array-method-this-argument`: Check `Array.from()`

### LINKED ISSUE #2144: `no-array-method-this-argument` doesn't warn about `Array.from()`
`no-array-method-this-argument` rule doesn't warn about `Array.from()`

```js
const foo = Array.from(bar, element => isUnicorn(element), baz);
```

Expected: ``Do not use the `this` argument in `Array#from()` `` warning in the above example for the `baz` argument.

Actual: no warning.

### PR DESCRIPTION
**Pull request by fisker**
Fixes #2144

### CODE REVIEW

File: test/snapshots/no-array-method-this-argument.mjs.md [hunk 1]
```diff
@@ -392,7 +568,30 @@ Generated by [AVA](https://avajs.dev).
...
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | Array.from(iterableOrArrayLike, (( foo as bar )), (( thisArgument )),)␊
+        |                                                      ^^^^^^^^^^^^ Do not use the \`this\` argument in \`Array.from()\`.␊
+    ␊
+    --------------------------------------------------------------------------------␊
+    Suggestion 1/2: Remove the second argument.␊
```
Comment (fisker) [inline test/snapshots/no-array-method-this-argument.mjs.md:586]: Note to myself: message need fix, it's the third, not the second argument.

**Comment by theSunDoesntExist** [comment]
Hey @fisker 
I believe you also need to add `Check.from()` in the documentation of this rule.
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-method-this-argument.md

> This rule checks following array methods accepts thisArg:
> 
>     Array#every()
>     Array#filter()
>     Array#find()
>     Array#findLast()
>     Array#findIndex()
>     Array#findLastIndex()
>     Array#flatMap()
>     Array#forEach()
>     Array#map()
>     Array#some()

**Comment by fisker** [comment]
> I believe you also need to add Check.from() in the documentation of this rule.

Missed that, would you like to fix it?

**Comment by theSunDoesntExist** [comment]
I don't know how to do it. Thanks though <3

**Comment by fisker** [comment]
No problem, I'll fix that.

**Pull request merged**